pkg/operator/controller/status/controller.go

```diff
@@ -168,23 +179,41 @@ func initializeClusterOperator(co *configv1.ClusterOperator) {
 	}
 }
 
+type operatorState struct {
+	Namespace          *corev1.Namespace
+	IngressControllers []operatorv1.IngressController
+	DNSRecords         []iov1.DNSRecord
+}
+
 // getOperatorState gets and returns the resources necessary to compute the
 // operator's current state.
-func (r *reconciler) getOperatorState(nsName string) ([]operatorv1.IngressController, *corev1.Namespace, error) {
+func (r *reconciler) getOperatorState(nsName string) (*operatorState, error) {
```

Comment:
Is there any reason `operatorState` should be a pointer value?

Reply:
No good reason — changed to a value type